megatron/rl/sequence_packing_utils.py

```diff
@@ -0,0 +1,1109 @@
+import torch
+import math
+import numpy as np
+from typing import List, Dict, Any, Tuple, Optional
+from torch.utils.data import DataLoader, TensorDataset
+from dataclasses import dataclass, field
+from megatron.core.utils import log_single_rank
+from megatron.training.global_vars import get_args, get_tokenizer
+from megatron.training.utils import get_nvtx_range
+from megatron.core.packed_seq_params import PackedSeqParams
+from megatron.core.num_microbatches_calculator import get_num_microbatches
+from megatron.core import mpu
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class PackingInfo:
+    """Information about how sequences are packed into bins.
+    
+    Attributes:
+        bin_seq_indices: List where each element contains the global sequence indices in that bin
+        seq_starts: Dict mapping bin index to list of start positions for each sequence in that bin
+        seq_lengths: List of all original sequence lengths (indexed by global sequence index)
+        seq_to_bin_idx: List mapping each global sequence index to its bin index
+        packing_algo: Algorithm used for distributing bins ('fifo' or 'round-robin')
+    """
+    bin_seq_indices: List[List[int]]
+    seq_starts: Dict[int, List[int]]
+    seq_lengths: List[int]
+    seq_to_bin_idx: List[Optional[int]]
+    packing_algo: str = 'fifo'
+
+
+@dataclass
+class PackingContext:
+    """Context containing all information needed for sequence packing during training.
+    
+    Attributes:
+        bin_size: Maximum size of each bin (in tokens)
```

Comment:
nit: let's sort the docstrings args the same way the dataclass fields are defined